components/BarcodeScanner/index.js

```diff
@@ -0,0 +1,85 @@
+import { useRef, useState } from "react";
+import useSWR from "swr";
+import styled from "styled-components";
+import Image from "next/image";
+import dynamic from "next/dynamic";
+const Scanner = dynamic(() => import("@/components/Scanner"), {
+  nossr: true,
+});
+
+const StyledSection = styled.section`
+  position: relative;
+  overflow: hidden;
+`;
+
+const StyledCanvas = styled.canvas`
+  position: absolute;
+  top: 0px;
+  height: 100%;
+`;
+
+const StyledImage = styled(Image)`
+  width: 25%;
+  height: 25%;
+  object-fit: cover;
+`;
+export default function BarcodeScanner() {
+  const [scanning, setScanning] = useState(true);
+  const [result, setResult] = useState(null);
+
+  const { data } = useSWR(
+    result ? `https://de.openfoodfacts.org/api/v0/product/${result}.json` : null
+  );
+
+  const scannerRef = useRef(null);
+
+  function handleDetected(_result) {
+    setResult(_result.codeResult.code);
```

Comment:
I know it's private variable with a dash `_` in front.
But it's still confusing to use result, because there is a result state.
Might as well rename it to `receivedResult`, `newResult` or something more expressive.

Reply:
i will think about renaming , but for now it is ok for me